src/Orleans/Logging/FileLogger.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions.Internal;
+using Orleans.Runtime;
+
+namespace Orleans.Logging
+{
+    /// <summary>
+    /// The log output which all <see cref="FileLogger"/> share to log messages to 
+    /// </summary>
+    public class FileLoggingOutput 
+    {
+        private readonly TimeSpan flushInterval = Debugger.IsAttached ? TimeSpan.FromMilliseconds(10) : TimeSpan.FromSeconds(1);
+        private DateTime lastFlush = DateTime.UtcNow;
+        private StreamWriter logOutput;
+        private readonly object lockObj = new object();
+        private string logFileName;
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="fileName"></param>
+        public FileLoggingOutput(string fileName)
+        {
+            logOutput = new StreamWriter(File.Open(fileName, FileMode.Append, FileAccess.Write, FileShare.Write));
+            this.logFileName = fileName;
+        }
+
+        /// <summary>
+        /// Log message for <see cref="FileLogger"/> instance whose category is <paramref name="category"/>
+        /// </summary>
+        /// <typeparam name="TState"></typeparam>
+        /// <param name="logLevel"></param>
+        /// <param name="eventId"></param>
+        /// <param name="state"></param>
+        /// <param name="exception"></param>
+        /// <param name="formatter"></param>
+        /// <param name="category"></param>
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception,
+            Func<TState, Exception, string> formatter, string category)
+        {
+            var logMessage = FormetMessage(DateTime.UtcNow, logLevel, category, formatter(state, exception), exception, eventId);
+            lock (this.lockObj)
+            {
+                if (this.logOutput == null) return;
+
+                this.logOutput.WriteLine(logMessage);
+                var now = DateTime.UtcNow;
+                if (now - this.lastFlush > flushInterval)
+                {
+                    this.lastFlush = now;
+                    this.logOutput.Flush();
+                }
+            }
+        }
+
+        private static string FormetMessage(
```

Comment:
There is a typo here. But wasn't this copy/pasted? The original code didn't have the typo, so I'm wondering if there are more changes...

Reply:
it was copied from Microsoft.Orleans.Logging.Legacy.OrleansLoggingUtils
u can do a diff checker online tool to findout the diff between them. Basically I removed "logging loggerType, ipEndpoint" from it.